mantidproject/mantid
# Script generation for colorfill plots

**Description of work.**
Added the ability to script our colorfill plots (for simple plots with only one colorfill plot, not group workspaces).

**To test:**
``` python # import mantid algorithms, numpy and matplotlib
from mantid.simpleapi import *
import matplotlib.pyplot as plt
import numpy as np

ws1 = CreateSampleWorkspace()
fig, ax = plt.subplots(subplot_kw={'projection':'mantid'})
c = ax.imshow(ws1)
cbar=fig.colorbar(c)
cbar.set_label('Intensity (arb. units)') #add text to colorbar
plt.show()

ws2 = CreateSampleWorkspace()
ws3 = CreateSampleWorkspace()
ws4 = CreateSampleWorkspace()
wsGroup = GroupWorkspaces([ws2,ws3,ws4])
```
1. open workbench and run this script.
1. the plot generated is not a standard workbench plot and should not have the script icon on the toolbar, if so close it.
1. Plot a colorfill plot from wsGroup, again no script icon
1. plot a colorfill plot from one of ws1-4,  You should get the script icon.  Script it to clipboard
1. Close the plot and script to a new script window
1. run the script you should get a matching plot.
1.  plot a colorfill plot from ws1, open the plot options and change some settings to make the plot look different. Script it to clipboard
1. Close the plot and script to a new script window
1. run the script you should get a matching plot.
1. Check scripting of line plots is still fine

Fixes #28600


---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
